packages/core/src/contracts/ownedAddressTCRWithAppeals.ts

```diff
@@ -0,0 +1,398 @@
+    if (!revealPeriodActive) {
+      isInRevealPhase = false;
+    }
+
+    return isInRevealPhase;
+  }
+
+  public async getTokenAddress(): Promise<EthAddress> {
+    return this.instance.token.callAsync();
+  }
+
+  /**
+   * Checks if an address has application associated with it
+   * App can either be in progress or approved
+   * @param address Address of potential listing to check
+   */
+  public async appWasMade(listingAddress: EthAddress): Promise<boolean> {
+    return this.instance.appWasMade.callAsync(listingAddress);
+  }
+
+  /**
+   * Checks if an address has application/listing has an unresolved challenge
+   * @param address Address of potential listing to check
+   */
+  public async challengeExists(listingAddress: EthAddress): Promise<boolean> {
+    return this.instance.challengeExists.callAsync(listingAddress);
+  }
+
+  /**
+   * Checks if an address has challenge that can be resolved
+   * @param address Address of potential listing to check
+   */
+  public async challengeCanBeResolved(listingAddress: EthAddress): Promise<boolean> {
+    return this.instance.challengeExists.callAsync(listingAddress);
+  }
+
+  /**
+   * Determines number of tokens to award winning party in challenge
+   * @param address ID of challenge to determine reward for
+   */
+  public async determineReward(challengeID: BigNumber): Promise<BigNumber> {
+    return this.instance.determineReward.callAsync(challengeID);
+  }
+
+  /**
+   * Gets whether or not specified user has a token claim for specified challenge
+   * @param challengeID ID of challenge to check
+   * @param address Address of voter to check
+   */
+  public async tokenClaims(challengeID: BigNumber, voter?: EthAddress): Promise<boolean> {
+    let who = voter;
+    if (!who) {
+      who = requireAccount(this.web3Wrapper);
+    }
+    return this.instance.tokenClaims.callAsync(challengeID, who);
+  }
+
+  /**
+   * Contract Transactions
+   */
+
+  /**
+   * Starts an application for a listing
+   * @param address Address of new listing
+   * @param deposit How many tokens to deposit to start application
+   * @param applicationContent Content associated with application
+   */
+  public async apply(
+    listingAddress: EthAddress,
+    deposit: BigNumber,
+    applicationContent: string,
+  ): Promise<{txHash: TxHash, awaitReceipt: Promise<CivilTransactionReceipt>}> {
+    const uri = await this.contentProvider.put(applicationContent);
+    return this.applyWithURI(listingAddress, deposit, uri);
+  }
+
+  /**
+   * Starts an application for a listing
+   * @param address Address of new listing
+   * @param deposit How many tokens to deposit to start application
+   * @param applicationContentURI (URI that points to data object)
+   */
+  public async applyWithURI(
+    listingAddress: EthAddress,
+    deposit: BigNumber,
+    applicationContentURI: string,
+  ): Promise<{txHash: TxHash, awaitReceipt: Promise<CivilTransactionReceipt>}> {
```

Comment:
Check latest branch in other wrappers for a more abstract way.

(Additionally this one might have a memory leakage error)